client/client.go

```diff
@@ -114,6 +114,16 @@ func (c Client) checkRoot() error {
 	if !bytes.Equal(hash[:], hashSha256) {
 		return fmt.Errorf("Cached root sha256 did not match snapshot root sha256")
 	}
+
+	root := &data.SignedRoot{}
+	err = json.Unmarshal(raw, root)
+	if err != nil {
+		return ErrCorruptedCache{file: "root.json"}
```

Comment:
All errors out of checkRoot get converted into ErrLocalRootExpired in the `update()` function above.

This should already be covered by the client tests but apparently we don't have any that include local expired data. Need to look at adding to them.